hashmap/utils.mbt

```diff
@@ -54,6 +54,26 @@ pub fn iter[K, V](self : T[K, V]) -> Iter[(K, V)] {
   )
 }
 
+///|
+pub fn T::iter2[K, V](self : T[K, V]) -> Iter2[K, V] {
+  Iter2::new(
+    fn(yield_) {
+      let len = self.entries.length()
+      for i = 0; i < len; i = i + 1 {
+        match self.entries[i] {
```

Comment:
is this the efficient way to iterate hashmap?

Reply:
This is copied from `T::iter`. Since `@hashmap.T` do not record keys in linked list as `@builtin.Map`, there's no efficient way to iterate over keys.